### DISCUSSION TITLE
how to exclude prisma clients from bundling

### DISCUSSION THREAD

**Discussion by arily** [nuxt/nuxt#16084] [Questions]
I have a setup which requires multiple generated prisma client, so I have to put it outside node_modules, in my case `.prisma`.

I have an file structure like this:

```
.prisma/
  A/
  B/
```
Then I can import the prisma client like this:
```js
import {
  PrismaClient,
} from '~~/.prisma/A/index'
```
dev works flawlessly but once built it stopped working.
```
[nuxt] [request error] [unhandled] [500] __dirname is not defined in ES module scope
  at ./server/chunks/_trpc_.mjs:1:443816  
  at ./server/chunks/_trpc_.mjs:1:766355  
  at ModuleJob.run (node:internal/modules/esm/module_job:193:25)  
  at async Promise.all (index 0)  
  at async ESMLoader.import (node:internal/modules/esm/loader:530:24)  
  at async Object.handler (./server/node_modules/h3/dist/index.mjs:681:19)  
  at async Server.toNodeHandle (./server/node_modules/h3/dist/index.mjs:745:7)
```

![image](https://user-images.githubusercontent.com/10007589/209212993-057dd184-83b2-4bbf-b6b4-1c7a117a0a61.png)
it looks like that the prisma client has been bundled into the server bundle.

this won't happen if I can put clients inside node_nodules but it looks like prisma does not support multiple client in the same default location.

**Comment by husayt** [comment]
rather than excluding it better to to tell nuxt to handle it commonjs. The question is how and I am trying to figure that out

**Reply by arily** [reply]
if you just want to import CJS you can use [createRequire](https://nodejs.org/api/module.html)

**Reply by arily** [reply]
https://github.com/prisma/prisma/issues/5030#issuecomment-1406125252
